lib/audit_test.pm

```diff
@@ -0,0 +1,133 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base module for audit-test test cases
+# Maintainer: llzhao <llzhao@suse.com>
+
+package audit_test;
+
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Mojo::File 'path';
+
+use base 'consoletest';
+
+our @EXPORT = qw(
+  $testdir
+  $testfile_tar
+  $baseline_file
+  $code_repo
+  $mode
+  run_testcase
+  parse_testcase_log
+  compare_run_log
+);
+
+our $testdir      = '/tmp/';
+our $code_repo    = get_var('CODE_BASE', 'https://gitlab.suse.de/security/audit-test-sle15/-/archive/master/audit-test-sle15-master.tar');
+our $testfile_tar = get_testsuite_name($code_repo);
+our $mode         = get_var('MODE', 64);
+
+# $current_file: current output file name; $baseline_file: baseline file name
+our $current_file  = 'run.log';
```

Comment:
There is a file named `rollup.log` after runing `run.bash`, it only records the test case and it result, like:
```
  [0] chmod__cap_file_priv_success_super                                                                                                           PASS
[1] chmod__cap_file_priv_fail_test                                                                                                                         PASS
[2] chown__cap_file_priv_success_super                                                                                                              PASS
```
 we could use this file to compare to the baseline file. Because `run.log` has more log messages.

Reply:
Yes, I noticed. It is ok too.
But IMO it still needs to parse the results:https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12727. Also it needs to upload the file and write more code for this. The parsing code writing in perl is less consuming than shift code to shell (e.g., assert_run()).